datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -439,6 +441,15 @@ func datadogSecurityMonitoringRuleSchema() map[string]*schema.Schema {
 			Description: "The rule type.",
 			Default:     "log_detection",
 		},
+		"validate": {
```

Comment:
Could we update the function signature to accept a boolean. Something along the lines of `datadogSecurityMonitoringRuleSchema(includeValidate bool)` and add validate block based on this. Mainly because this schema is reused in the data-source which doesn't make use of this